src/elf/mod.rs

```diff
@@ -187,23 +187,26 @@ mod impure {
 
             let section_headers = SectionHeader::parse(bytes, header.e_shoff as usize, header.e_shnum as usize, ctx)?;
 
-            let strtab_idx = header.e_shstrndx as usize;
-            let shdr_strtab = if strtab_idx >= section_headers.len() {
-                Strtab::default()
-            } else {
-                let shdr = &section_headers[strtab_idx];
-                try!(Strtab::parse(bytes, shdr.sh_offset as usize, shdr.sh_size as usize, 0x0))
+            let get_strtab = |section_headers: &[SectionHeader], section_idx: usize| {
+                if section_idx >= section_headers.len() {
+                    Ok(Strtab::default())
```

Comment:
Hmm I've been thinking about this and it might be better to return an error here, or log, I'm leaning towards latter but:

My reasoning is that it's a pretty serious error if the section index is oob, and returning a default strtab will just return empty strings for any references, further exacerbating debuggging. 

On the other hand the program is still technically executable without any errors whatsoever, so it might be worth parsing and warning.

What do you think ?